src/main/java/com/exadel/discount/service/impl/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package com.exadel.discount.service.impl;
+
+import com.exadel.discount.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository repository;
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        log.debug("getting a user data from a database");
+        com.exadel.discount.entity.User user = repository.findByEmail(email);
+        log.debug("successfully getting a user data from a database");
```

Comment:
too much logs

Reply:
This is done locally.

Reply:
It was resolved in commit "Commit with amendments №2".